### DISCUSSION TITLE
Any way to change contents of navigator.userAgent

### DISCUSSION THREAD

**Discussion by drbsoftware** [MicrosoftEdge/WebView2Feedback#4351] [Q&A] [answered]
navigator.userAgent is read-only. I cannot tell from the documentation on WebViewSettings.UserAgent whether the setting will apply to navigator.userAgent.  We have JavaScript loaded from the website which reads navigator.userAgent to determine the device.

If this will work, could the documentation be updated to reflect this use case?

**Comment by victorhuangwq** [comment]
@yildirimcagri-msft is navigator.userAgent read-only?

**Reply by ghost** [reply, marked as answer]
WebView2Settings.UserAgent should also set navigator.userAgent. Let us know if you are seeing a different behavior.
